### DISCUSSION TITLE
Is there a way of having an index for the current step ?

### DISCUSSION THREAD

**Discussion by Galand-V** [getgauge/gauge#2176] [General]
Hello Everyone, 

I am currently using the same step in a single testcase and I would like to have a different behaviour for this step depending on an index. 

Example : 

* first step
* get value from a json
* second step
* get value from a json
* check values from the json

----------------- 

step("get value from a json", async () => {
// I would like to have 2 values in my dataStore with the same step
gauge.dataStore.scenarioStore.put('propertyName${index}', response.value);
});

Do you have any ideas ? 

I look through the documentation and did not get anything ont this subject. 

Thank you ! 

Victor

**Comment by sriv** [comment]
rather than depending on the index, would it be possible to pass a parameter that can be used to lookup the respective value? The challenge with the index approach that I can expect:

- the same step cannot be reused in another scenario without ensuring that it is in the same index
- the scenario cannot be enhanced/modified if it breaks the step indices (and there is no clean way to warn about this).
